### DISCUSSION TITLE
Failed to distribute generation in batches

### DISCUSSION THREAD

**Discussion by Xellosdark12** [papuSpartan/stable-diffusion-webui-distributed#27] [General]
Setup
- I am using automatic 1111 created on docker of unraid with two display cards
- my objective is to set up 2nd SD instance to speed up the image generation
- After the setup, both instances look good in benchmark check; however, error is returned when I submit 4 (size) * 3 (batch) job as follows
- My master instance could not be distributed with any image creation (with error), any clue suggested?

Thanks,
    Francis 

======
DISTRIBUTED | INFO     World composition:                           world.py:260
                       1. 'test'(192.168.12.12:9100) - 14.98 ipm                
                       2. 'master'(0.0.0.0:9000) - 8.21 ipm                     
                       total: ~23.20 ipm                                        
DISTRIBUTED | INFO     Job distribution:                            world.py:554
                       4 * 3 iteration(s): 4 images total                       
                       'master' - 0 image(s) @ 8.21 ipm                         
                       'test' - 12 image(s) @ 14.98 ipm                         
                                                                                
*** Error completing request
*** Arguments: ('task(zersw1ob58czuhe)', <gradio.routes.Request object at 0x15477a92bf10>, '(masterpiece, high quality:1.4), white bikini, natural skin, behind is beach, 8k, portrait, best quality, ultra high res, anime, (solo, 1 girl), extremely detailed lighting, cinematic lighting, soft lights, realistic, (finely detailed beautiful eyes), <lora:ayukawa_madoka_v2-10:0.9>, (full body)', 'paintings, cartoon, rendered, sketches, (worst quality:2), (low quality:2), (normal quality:2), child, chest hair, mole, lowres, normal quality, monochrome, grayscale, lowres, text, error, cropped, jpeg artifacts, ugly, duplicate, morbid, mutilated, out of frame, extra fingers, mutated hands, poorly drawn hands, poorly drawn face, mutation, deformed, blurry, dehydrated, bad anatomy, bad proportions, extra limbs, cloned face, disfigured, gross proportions, ((malformed limbs, missing arms, missing legs, extra arms, extra legs, fused fingers, too many fingers)), long neck, username, watermark, signature, ng_deepnegative_v1_75t, EasyNegative, bad-hands-5, bad-artist-anime', [], 3, 4, 7, 1024, 1024, False, 0.7, 2, 'Latent', 0, 0, 0, 'Use same checkpoint', 'Use same sampler', 'Use same scheduler', '', '', [], 0, 83, 'DDIM', 'Karras', False, '', 0.8, -1, False, -1, 0, 0, 0, False, '', '', '', 'Refresh 🔄', '🆕 Refresh checkpoints', '⚙️ Run script', '📜 Reload config', '📊 Redo benchmarks', '⏸️ Interrupt all', '📊 Redo benchmarks', '🔁 Restart All', '🔌 Reconnect workers', '', '', '', False, False, '', 0.0, False, '', '', 'Update API Credentials', 'Add/Update Worker', 'Remove Worker', False, 3.0, True, 'Update', False, False, 'positive', 'comma', 0, False, False, 'start', '', 1, '', [], 0, '', [], 0, '', [], True, False, False, False, False, False, False, 0, False) {}
    Traceback (most recent call last):
      File "/config/02-sd-webui/webui/modules/call_queue.py", line 57, in f
        res = list(func(*args, **kwargs))
                   ^^^^^^^^^^^^^^^^^^^^^
      File "/config/02-sd-webui/webui/modules/call_queue.py", line 36, in f
        res = func(*args, **kwargs)
              ^^^^^^^^^^^^^^^^^^^^^
      File "/config/02-sd-webui/webui/modules/txt2img.py", line 109, in txt2img
        processed = processing.process_images(p)
                    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
      File "/config/02-sd-webui/webui/modules/processing.py", line 845, in process_images
        res = process_images_inner(p)
              ^^^^^^^^^^^^^^^^^^^^^^^
      File "/config/02-sd-webui/webui/modules/processing.py", line 893, in process_images_inner
        p.setup_prompts()
      File "/config/02-sd-webui/webui/modules/processing.py", line 1450, in setup_prompts
        super().setup_prompts()
      File "/config/02-sd-webui/webui/modules/processing.py", line 441, in setup_prompts
        self.main_prompt = self.all_prompts[0]
                           ~~~~~~~~~~~~~~~~^^^
    IndexError: list index out of range

**Comment by papuSpartan** [comment]
What commit/release of webui are you on for this?

**Comment by papuSpartan** [comment]
If you make your main instance use the faster device then you shouldn't have this issue.

I know why this is happening but I can't have it fixed well immediately.
